devnet-sdk/testing/systest/testing_test.go

```diff
@@ -105,22 +105,58 @@ func (m *mockChain) SupportsEIP(ctx context.Context, eip uint64) bool {
 func (m *mockChain) Config() (*params.ChainConfig, error) {
 	return nil, fmt.Errorf("not implemented on mockChain")
 }
-func (m *mockChain) Addresses() descriptors.AddressMap {
-	return descriptors.AddressMap{}
+func (m *mockChain) Addresses() system.AddressMap {
+	return system.AddressMap{}
+}
+
+// mockL2Chain implements a minimal system.L2Chain for testing
+type mockL2Chain struct{}
```

Comment:
Can we embed `type mockChain` into `type mockL2Chain` for reusing all the mock methods?
```go
type mockL2Chain struct {
    mockChain
}
```

Reply:
Fixed in 2895f546